test/add-member.test.js

```diff
@@ -272,3 +272,96 @@ test('addMembers too many members', async (t) => {
 
   await p(alice.close)(true)
 })
+
+test('can or cannot add someone back into a group', async (t) => {
+  const alice = Testbot({
+    keys: ssbKeys.generate(null, 'alice'),
+    mfSeed: Buffer.from(
+      '000000000000000000000000000000000000000000000000000000000000a1ce',
+      'hex'
+    ),
+  })
+  let bob = Testbot({
+    name: 'bobrestart',
+    keys: ssbKeys.generate(null, 'bob'),
+    mfSeed: Buffer.from(
+      '0000000000000000000000000000000000000000000000000000000000000b0b',
+      'hex'
+    ),
+  })
+
+  await Promise.all([alice.tribes2.start(), bob.tribes2.start()])
+
+  const bobRoot = await p(bob.metafeeds.findOrCreate)()
+
+  await replicate(alice, bob).catch(t.error)
+
+  const { id: groupId } = await alice.tribes2
+    .create()
+    .catch((err) => t.error(err, 'alice failed to create group'))
+
+  await alice.tribes2
+    .addMembers(groupId, [bobRoot.id])
+    .then(() => t.pass('added bob'))
+    .catch((err) => t.error(err, 'add bob fail'))
+
+  await replicate(alice, bob).catch(t.error)
+
+  await bob.tribes2.acceptInvite(groupId)
+
+  await replicate(alice, bob).catch(t.error)
+
+  await alice.tribes2
+    .excludeMembers(groupId, [bobRoot.id])
+    .then(() => t.pass('alice excluded bob'))
+    .catch((err) => t.error(err, 'remove member fail'))
+
+  await replicate(alice, bob).catch(t.error)
+
+  await alice.tribes2
+    .addMembers(groupId, [bobRoot.id])
+    .then(() => t.pass('added bob back in again'))
+    .catch((err) => t.error(err, 'add bob back fail'))
+
+  await replicate(alice, bob).catch(t.error)
+
+  // TODO: test listinvite
+  await bob.tribes2.acceptInvite(groupId).catch(t.error)
+
+  async function verifyInGroup(peer) {
+    // TODO: test listInvites
+
+    await peer.tribes2
+      .acceptInvite(groupId)
+      .then(() => t.fail('consumed invite twice'))
+      .catch(() => t.pass("can't consume invite twice"))
+
+    const list = await pull(peer.tribes2.list(), pull.collectAsPromise())
+    t.equal(list.length, 1, 'one group')
+    t.equal(list[0].id, groupId, 'id in list is correct')
+
+    // TODO: test if writeKey is there
+
+    // TODO: test if `excluded` is there
+  }
+
+  await verifyInGroup(bob)
+
+  await p(bob.close)(true).then(() => t.pass("bob's client was closed"))
+  bob = Testbot({
+    rimraf: false,
+    name: 'bobrestart',
+    keys: ssbKeys.generate(null, 'bob'),
+    mfSeed: Buffer.from(
+      '0000000000000000000000000000000000000000000000000000000000000b0b',
+      'hex'
+    ),
+  })
+  t.pass('bob got a new client')
+  await bob.tribes2.start().then(() => t.pass('bob restarted'))
+
+  await verifyInGroup(bob)
```

Comment:
Fails here now because on restart we re-exclude ourselves. As mentioned in the todo in start() we need to only look for exclude messages in tip epochs. Although that function is in https://github.com/ssbc/ssb-tribes2/pull/106/